packages/smooth_app/lib/main.dart

```diff
@@ -66,36 +67,45 @@ class _SmoothAppState extends State<SmoothApp> {
   }
 
   Future<void> _init() async {
-    Function debugPrintAndRethrow(String message) => (dynamic error) {
-          debugPrint('$message: $error');
-          return error;
-        };
-    final Brightness brightness =
-        SchedulerBinding.instance?.window.platformBrightness ??
-            Brightness.light;
-    systemDarkmodeOn = brightness == Brightness.dark;
-    _userPreferences = await UserPreferences.getUserPreferences();
-    _productPreferences = ProductPreferences(ProductPreferencesSelection(
-      setImportance: _userPreferences.setImportance,
-      getImportance: _userPreferences.getImportance,
-      notify: () => _productPreferences.notifyListeners(),
-    ));
-    await _productPreferences
-        .loadReferenceFromAssets(DefaultAssetBundle.of(context))
-        // this is problematic - we should always be able to load the default
-        .catchError(debugPrintAndRethrow('Could not load reference files'));
-    await _userPreferences.init(_productPreferences);
-    _localDatabase = await LocalDatabase.getLocalDatabase()
-        // this is problematic - we should always be able to init the database
-        .catchError(debugPrintAndRethrow('Cannot init database'));
-    _themeProvider = ThemeProvider(_userPreferences);
+    try {
+      final Brightness brightness =
+          SchedulerBinding.instance?.window.platformBrightness ??
+              Brightness.light;
+      systemDarkmodeOn = brightness == Brightness.dark;
+      _userPreferences = await UserPreferences.getUserPreferences();
+      _productPreferences = ProductPreferences(ProductPreferencesSelection(
+        setImportance: _userPreferences.setImportance,
+        getImportance: _userPreferences.getImportance,
+        notify: () => _productPreferences.notifyListeners(),
+      ));
+      await _productPreferences
+          .loadReferenceFromAssets(DefaultAssetBundle.of(context));
+      // we should always be able to load the default
+      await _userPreferences.init(_productPreferences);
+      // we should always be able to init the database
+      _localDatabase = await LocalDatabase.getLocalDatabase();
+      _themeProvider = ThemeProvider(_userPreferences);
+    } catch (e) {
+      _initFatalErrorMessage = e.toString();
+    }
   }
 
   @override
   Widget build(BuildContext context) {
     return FutureBuilder<void>(
       future: _initFuture,
       builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (_initFatalErrorMessage != null) {
```

Comment:
Could we use `snapshot.hasError` and `snapshot.error` instead?

Reply:
@slava-sh Looks like a good idea. I'm about to commit.